filebeat/input/log/harvester.go

```diff
@@ -565,17 +569,25 @@ func (h *Harvester) newLogFileReader() (reader.Reader, error) {
 
 	if h.config.DockerJSON != nil {
 		// Docker json-file format, add custom parsing to the pipeline
-		r = readjson.New(r, h.config.DockerJSON.Stream, h.config.DockerJSON.Partial, h.config.DockerJSON.ForceCRI, h.config.DockerJSON.CRIFlags)
+		keepOriginalDocker = h.config.DockerJSON.KeepOriginal
+		r = readjson.New(r,
+			h.config.DockerJSON.Stream,
+			h.config.DockerJSON.Partial,
+			h.config.DockerJSON.ForceCRI,
+			h.config.DockerJSON.CRIFlags,
+			h.config.DockerJSON.KeepOriginal)
```

Comment:
5 Params already. Maybe we better introduce a configuration/settings struct to the readjson package.